wpimath/src/main/java/edu/wpi/first/math/kinematics/SwerveModuleState.java

```diff
@@ -51,8 +50,9 @@ public SwerveModuleState(double speedMetersPerSecond, Rotation2d angle) {
    * @param speed The speed of the wheel of the module.
    * @param angle The angle of the module.
    */
-  public SwerveModuleState(Measure<Velocity<Distance>> speed, Rotation2d angle) {
-    this(speed.in(MetersPerSecond), angle);
+  public SwerveModuleState(
+      Measure<? extends PerUnit<DistanceUnit, TimeUnit>> speed, Rotation2d angle) {
```

Comment:
Just out of curiosity, is there some reason this can't be `Measure<LinearVelocityUnit>` or maybe `Measure<VelocityUnit<DistanceUnit>>`?

Reply:
Because of Java generic interoperability. This constructor accepts a measure of any unit that extends from `PerUnit<DistanceUnit, TimeUnit` - which means `LinearVelocity`, `Per<DistanceUnit, TimeUnit>`, or `Velocity<DistanceUnit>` would all be acceptable input types

Reply:
The downside to wildcard bounds is that you can't use `in`, since the compiler doesn't know what _specific_ subtype will be stored in the measure